src/ApplicationInsights.Kubernetes/K8sHttpClient/KubeHttpSettingsWinContainerProvider.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    internal class KubeHttpSettingsWinContainerProvider : KubeHttpClientSettingsBase, IKubeHttpClientSettingsProvider
+    {
+        private readonly string _tokenFilePath;
+        private readonly string _certFilePath;
+        private readonly string _namespaceFilePath;
+
+        public KubeHttpSettingsWinContainerProvider(
+            ILogger<KubeHttpSettingsWinContainerProvider> logger,
+            string serviceAccountFolder = @"C:\var\run\secrets\kubernetes.io\serviceaccount",
+            string tokenFileName = "token",
+            string certFileName = "ca.crt",
+            string namespaceFileName = "namespace",
+            string kubernetesServiceHost = null,
+            string kubernetesServicePort = null)
+            : base(kubernetesServiceHost, kubernetesServicePort, logger)
+        {
+            // Container id won't be fetched for windows container.
+            ContainerId = null;
+            DirectoryInfo serviceAccountDirectory =
+                new DirectoryInfo(Arguments.IsNotNullOrEmpty(serviceAccountFolder, nameof(serviceAccountFolder)));
+            foreach (FileSystemInfo fileSystemInfo in serviceAccountDirectory.EnumerateFileSystemInfos("*", SearchOption.AllDirectories))
```

Comment:
Since you are interested in files only, would `EnumerateFiles()` be sufficient?

Reply:
Actually, why enumerate at all? Can you just try testing for the 3 desired file names explicitly?

Reply:
Good catch. I'll update the code. Got distracted by the investigation on filtering out the symbolic link.

Reply:
Roger